### PR TITLE
finished week-1 assignment of NodeJS

### PR DESCRIPTION
**Pull request by choudhary-keshav**
Pull Request Questionnaire

(1) Which Jira ticket(s) is this pull request for, and is the ticket now complete?

(2) Briefly describe your change.
I have pushed my week-1 assignment in this joint project. In my week-1 assignment I have added all daily assignments as well as main assignment for this week. I have used nodeJS modules like Express, http, fs.

(3) What parts of the application are affected by your change?
Nothing

(4) How did you test your change? If relevant, which role(s) and/or customer configuration(s) did you use?
Tested my application on various edge cases and some common scenarios

### CODE REVIEW

File: src/Week1/Daily Assignments/assignment1.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require("dotenv").config();
+
+import express, { Request, Response } from "express";
+const app = express();
+
+const fetchData = async () => {
```
Comment (aditya-s-19) [inline src/Week1/Daily Assignments/assignment1.ts]: define the return type

File: src/Week1/Daily Assignments/assignment1.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    return data;
+  } catch (error) {
+    console.log("Error in fetching data from API");
+    return null;
+  }
+};
+
+app.get("/", async (req: Request, res: Response) => {
+  const data = await fetchData();
+  if (data !== null) {
```
Comment (aditya-s-19) [inline src/Week1/Daily Assignments/assignment1.ts]: shorten this condition

File: src/Week1/Daily Assignments/assignment3.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  }
+});
+
+app.post("/:postId?", async (req: Request, res: Response) => {
+  const { postId } = req.params;
+  const data = await fetchData();
+
+  if (data !== null && postId) {
+    const filteredData = data.filter(
+      (item: any) => item.postId === parseInt(postId)
```
Comment (RitikaGarg-ITT) [inline src/Week1/Daily Assignments/assignment3.ts]: can you explain the reason of type any

File: src/Week1/Daily Assignments/assignment2.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  req.on("end", () => {
+    const receivedObject = JSON.parse(data);
+
+    receivedObject.lastName = "Choudhary";
+
+    const modifiedObject = JSON.stringify(receivedObject);
+
+    res.write(modifiedObject + "\n\n");
+
```
Comment (shreyansh-khatri) [inline src/Week1/Daily Assignments/assignment2.ts]: please remove extra spaces above

File: src/Week1/Daily Assignments/assignment3.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
+require("dotenv").config();
+
+import express, { Request, Response } from "express";
+const app = express();
+
+const fetchData = async () => {
```
Comment (shreyansh-khatri) [inline src/Week1/Daily Assignments/assignment3.ts]: please specify return type

**Review by aditya-s-19 — APPROVED** [review]

**Review by shreyansh-khatri — APPROVED** [review]

**Review by RitikaGarg-ITT — APPROVED** [review]

**Review by shreegovinda — APPROVED** [review]

**Review by shreyansh-khatri — APPROVED** [review]

**Review by shreyansh-khatri — APPROVED** [review]
